test/e2e/scaling_dbsize_test.go

```diff
@@ -0,0 +1,66 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdenvvar"
+	"github.com/openshift/cluster-etcd-operator/test/e2e/framework"
+	"github.com/openshift/library-go/test/library"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	operandNameSpace  = "openshift-etcd"
+	etcdCRName        = "cluster"
+	etcdLabelSelector = "app=etcd"
+	quotaEnvName      = "ETCD_QUOTA_BACKEND_BYTES"
+	etcdPodName       = "etcd"
+)
+
+func TestEtcdDBScaling(t *testing.T) {
+	dbSize32GB := int32(32)
+	opClientSet := framework.NewOperatorClient(t)
+	clientSet := framework.NewClientSet("")
+	etcdPodsClient := clientSet.CoreV1Interface.Pods(operandNameSpace)
+
+	etcdCR, err := opClientSet.OperatorV1().Etcds().Get(context.Background(), etcdCRName, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	etcdCR.Spec.BackendQuotaGiB = dbSize32GB
+	etcdCR, err = opClientSet.OperatorV1().Etcds().Update(context.Background(), etcdCR, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	err = library.WaitForPodsToStabilizeOnTheSameRevision(t, etcdPodsClient, etcdLabelSelector, 3, 10*time.Second, 5*time.Second, 30*time.Minute)
```

Comment:
thanks @lance5890 

yes I had previous checks that was waiting for the new condition 

```
// wait for new etcd pods to be rolled out
	err = wait.PollUntilContextTimeout(context.Background(), time.Second, 10*time.Minute, true, func(ctx context.Context) (done bool, err error) {
		etcdPods, err := clientSet.Pods(operandNameSpace).List(context.Background(), metav1.ListOptions{LabelSelector: etcdLabelSelector})
		if err != nil {
			if apierrors.IsNotFound(err) {
				return false, nil
			}
			return false, err
		}

		for _, pod := range etcdPods.Items {
			if pod.Status.Phase == v1.PodRunning && time.Now().Sub(pod.CreationTimestamp.Time) < 1*time.Minute {
				return true, nil
			}
		}
		return false, nil
	})
	require.NoError(t, err)
```

but using `library-go` is always best .. 

the issue is that the helpers from library-go does not wait for new-revision, i am adding a helper now that does exactly that 👍🏽

Reply:
I'm not sure why we need another helper for this. Just increase the poll interval and the success threshold.

Reply:
The helper makes sure that a new revision has been rolled out before assertion 

the current e2e failure is due to asserting on the old etcd-pods

Reply:
it is entirely unnecessary, if you make the poll timeout 1 minute, have it finish for five successful runs it will still pass your test. Why does it take so long to update the env vars?

Reply:
ok let me try 👍🏽